docs/en/plugins/README.md

```diff
@@ -0,0 +1,413 @@
+
+##### 1. Tell Loomio to enable our plugin (important, don’t skip!)
+
+We’ll need to tell Loomio when to run this plugin code, otherwise it’ll skip everything and the plugin will be ‘off’ all the time. To do this, we can either provide an ENV variable to look at (the plugin will be enabled if the ENV is present)
+
+```
+# plugin.rb (within the `setup!` method)
+plugin.enabled = ‘ENABLE_THREAD_TAGGING’
+# (will be enabled when ENV[‘ENABLE_THREAD_TAGGING’] is truthy)
+```
+
+or, simply set to it ‘true’ all the time
+```
+# plugin.rb (within the `setup!` method)
+plugin.enabled = true
+```
+
+##### 2. Create two database tables for our new many-to-many relationship
+
+```
+# plugin.rb (within the `setup!` method)
+plugin.use_database_table :tag do |table|
+  table.belongs_to :group
+  table.string :name
+  table.string :color
+  table.timestamps
+end
+
+plugin.use_database_table :discussion_tag do |table|
+  table.belongs_to :tag
+  table.belongs_to :discussion
+  table.timestamps
+end
+```
+
+##### 3. Create a model to define what a tag does, and tell the plugin to use it
+
+```
+# plugins/loomio_tags/models/tag.rb
+class Tag < ActiveRecord::Base
+  belongs_to :group, dependent: :destroy
+  has_many :discussion_tags, dependent: :destroy
+  has_many :discussions, through: :discussion_tags
+
+  validates :group, presence: true
+  validates :name, presence: true
+  validates :color, presence: true
+end
+```
+```
+# plugins/loomio_tags/models/discussion_tag.rb
+class DiscussionTag < ActiveRecord::Base
+  belongs_to :discussion
+  belongs_to :tag
+
+  validates :discussion, presence: true
+  validates :tag, presence: true
+end
+```
+```
+# plugins/loomio_tags/plugin.rb (within the setup! method)
+plugin.use_class ‘models/tag’
+plugin.use_class ‘models/discussion_tag’
+```
+
+NB: We could also tell the plugin to load all of the files in the ‘models’ directory in a single line, like this:
+```
+plugin.use_class_directory :models
+```
+
+##### 4. Modify the existing Discussion and Group models to let it know about tags.
+
+```
+# plugins/loomio_tags/plugin.rb (within the setup! method)
+plugin.extend_class Discussion do
+  has_many :discussion_tags, dependent: :destroy
+  has_many :tags, through: :discussion_tags
+end
+
+plugin.extend_class Group do
+  has_many :tags, dependent: :destroy
+end
+```
+
+_Note: the commands which ‘plugin’ accepts are well-documented in the [README](https://github.com/loomio/loomio/tree/master/plugins) in our repository. If you’re trying to figure out how to do something, that’s a really great place to start._
+
+At this point, running `rails c` again should add those new tables and pop that new code into our instance.
+
+![](images/tables_exist.png)
+
+And if we run a few little experiments, we should see that we can now add tags and associate them with discussions via the console
+
```

Comment:
And if we run a few little experiments, we should see that we can now add tags and associate them with discussions via the console**:** [add colon]